test/stateful/ica-event-pipe/icaEventPipeConstruct.test.ts

```diff
@@ -38,22 +41,47 @@ describe('IcaEventPipeConstruct', () => {
   });
 
   test('SQS Queue configured with SSL', () => {
+    // make sure all QueuePolicies enforce SSL
     template.resourceCountIs('AWS::SQS::QueuePolicy', 2);
-    template.hasResourceProperties('AWS::SQS::QueuePolicy', {
-      PolicyDocument: Match.objectLike({
-        Statement: [
-          {
-            Action: 'sqs:*',
-            Effect: 'Deny',
-            Condition: {
-              Bool: {
-                'aws:SecureTransport': 'false',
+    template.allResourcesProperties(
+      'AWS::SQS::QueuePolicy',
+      Match.objectLike({
+        PolicyDocument: {
+          Statement: Match.arrayWith([
+            Match.objectLike({
+              Action: 'sqs:*',
+              Effect: 'Deny',
+              Condition: {
+                Bool: {
+                  'aws:SecureTransport': 'false',
+                },
               },
-            },
-          },
-        ],
-      }),
-    });
+            }),
+          ]),
+        },
+      })
+    );
+  });
+
+  test('SQS Queue configured for ICA events', () => {
+    template.hasResourceProperties(
+      'AWS::SQS::QueuePolicy',
+      Match.objectLike({
+        PolicyDocument: {
+          Statement: Match.arrayWith([
+            Match.objectLike({
+              Action: Match.arrayWith(['sqs:SendMessage']),
+              Effect: 'Allow',
+              Principal: {
+                AWS: {
+                  'Fn::Join': ['', ['arn:', Match.anyValue(), ':iam::' + icaTestAccount + ':root']],
+                },
+              },
+            }),
+          ]),
+        },
+      })
+    );
```

Comment:
Auuu, Nice! 👍 Very comprehensively testing.

Reply:
That was where I spent most of the time ;-)